python_modules/libraries/dagster-components/dagster_components/lib/definitions_component.py

```diff
@@ -44,7 +45,24 @@ def load(cls, context: ComponentLoadContext) -> Self:
 
     def build_defs(self, context: ComponentLoadContext) -> Definitions:
         with pushd(str(context.path)):
-            module = import_uncached_module_from_path("definitions", str(self.definitions_path))
+            # Inserting the current directory path into the path
+            # get absolute paths to work
+            here = context.path.absolute()
+            if here not in sys.path:
+                sys.path.insert(0, str(here))
+
+            # This only way to get package-style relative imports to work
+            # is to name the module directory_name.module_name and then
+            # have __init__.py in the directory_name directory and then
+            # insert that directory into the path
+            here = context.path.parent.absolute()
+            if here not in sys.path:
+                sys.path.insert(0, str(here))
+
+            abs_definitions_path = (context.path / self.definitions_path).absolute()
+            module_name = f"{abs_definitions_path.parent.name}.{abs_definitions_path.stem}"
+
+            module = import_module_from_path(module_name, str(self.definitions_path))
```

Comment:
@smackesey per your point this is how I had to do this to get relative and absolute imports to work but it seems..... fishy